libcaf_core/caf/flow/op/from_steps.hpp

```diff
@@ -215,12 +207,22 @@ class from_steps_sub : public detail::plain_ref_counted,
 
   void run_later() {
     if (!running_) {
-      running_ = true;
-      ctx_->delay_fn([ptr = strong_this()] { ptr->do_run(); });
+      ctx_->delay_fn([ptr = strong_this()] { ptr->do_run_impl(); });
     }
   }
 
   void do_run() {
+    // Note: external references to this object may be discarded while running
+    //       do_run_impl. Hence, we need to make sure that the object is alive
+    //       during the entire execution of do_run_impl. Otherwise, we might
+    //       trigger a heap-use-after-free.
+    ref();
+    auto guard = detail::make_scope_guard([this] { deref(); });
```

Comment:
Was the intent here to increment the ref count by taking `this`, and decrement it by `deref`?